Source/Fuse.Scripting/ScriptClass.uno

```diff
@@ -105,25 +112,107 @@ namespace Fuse.Scripting
 			_method = method;
 		}
 
+		[Obsolete("Use ScriptMethod<T>(string, Uno.Func<Fuse.Scripting.Context, T, object[], object)>) instead")]
+		public ScriptMethod(string name, Func<Context, T, object[], object> method, ExecutionThread thread): this(name, method, thread, true)
+		{
+		}
+
+		/** Create a ScriptMethod that will run on the script-thread
+
+			@param name Name of method
+			@param method The native implementation of the method
+		*/
+		public ScriptMethod(string name, Func<Context, T, object[], object> method): this(name, method, ExecutionThread.JavaScript, true)
+		{
+		}
+
+		[Obsolete("Use ScriptMethod<T>(string, Uno.Action<T)>), ScriptMethod<T>(string, Uno.Action<T, object[])>) or ScriptMethod<T>(string, Uno.Func<Fuse.Scripting.Context, T, object[], object)>) instead")]
 		public ScriptMethod(string name, Action<Context, T, object[]> method, ExecutionThread thread): base(name, thread)
 		{
 			if (method == null)
 				throw new ArgumentNullException(nameof(method));
 
+			_oldVoidMethod = method;
+		}
+
+		/** Create an argument-less ScriptMethod that will run on the UI-thread
+
+			@param name Name of method
+			@param method The native implementation of the method
+		*/
+		public ScriptMethod(string name, Action<T> method): base(name, ExecutionThread.MainThread)
+		{
+			if (method == null)
+				throw new ArgumentNullException(nameof(method));
+
 			_voidMethod = method;
 		}
 
+		/** Create a ScriptMethod that will run on the UI-thread
+
+			@param name Name of method
+			@param method The native implementation of the method
+		*/
+		public ScriptMethod(string name, Action<T, object[]> method): base(name, ExecutionThread.JavaScript)
+		{
+			if (method == null)
+				throw new ArgumentNullException(nameof(method));
+
+			_method = new ArgumentMirrorClosure<T>(method).Run;
+		}
+
+		class ArgumentMirrorClosure<T>
+		{
+			readonly Action<T, object[]> _action;
+			public ArgumentMirrorClosure(Action<T, object[]> action)
+			{
+				_action = action;
+			}
+
+			public object Run(Context c, T obj, object[] args)
+			{
+				var marahalledArguments = new List<object>();
+				foreach (var arg in args)
+					marahalledArguments.Add(c.Reflect(arg));
+
+				UpdateManager.PostAction(new CallWithArgumentsClosure(_action, obj, marahalledArguments.ToArray()).Run);
```

Comment:
I wonder if we should allow PostAction to also take an IAction object? Any place using closure classes could then implement it and we'd save an allocation of the action.

Reply:
I'm not sure what that would solve. I could also just stuff the arguments inside ArgumentMirrorClosure<T>, and add a separate method on the UI-thread to save the allocation...

Reply:
I'm not sure I understand. We need an allocation per invocation as well. What allocation would be avoided?

Reply:
The `.Run` bit creates a binding object implicitly. If we added a `IAction` interface we'd avoid that overhead.

In profiling the creation of these action bindings does appear a lot. However, in the JS->Uno functions there's already so much overhead it probably won't be noticed.

Reply:
Ah, I see. Yeah, perhaps there's some use-cases where this would make sense, but I doubt this is that one.